collection.go

```diff
@@ -73,11 +73,15 @@ func (c Collection) PrimaryValues() []interface{} {
 	if index != nil {
 		for i := range index {
 			var (
-				values = make([]interface{}, c.rv.Len())
+				idxLen = c.rv.Len()
+				values = make([]interface{}, 0, idxLen)
 			)
 
-			for j := range values {
-				values[j] = c.rvIndex(j).Field(index[i]).Interface()
+			for j := 0; j < idxLen; j++ {
+				if item := c.rv.Index(j); item.Kind() == reflect.Ptr && item.IsNil() {
+					continue
+				}
+				values = append(values, c.rvIndex(j).Field(index[i]).Interface())
```

Comment:
to prevent double calls on rv.Index, how about something like this:
```suggestion
				if item := c.rvIndex(j); item.IsValid() {
					values = append(values, item.Field(index[i]).Interface())
				}
```

https://play.golang.org/p/Os_66nnUyUm

Reply:
seems to do the job, fixed